viewing_party/party.py

```diff
@@ -1,23 +1,212 @@
 # ------------- WAVE 1 --------------------
 
+from enum import unique
+from itertools import count
+
+# creates a movie object with a given title, genre, and rating
+# parameters: title - title of movie, genre - genre of movie, rating - rating of the movie 
 def create_movie(title, genre, rating):
-    pass
+    # we first check if all of our arguments are truthy if not we return None
+    if not title or not genre or not rating:
+        return None
+    # we then create the dictionary with the keys title, genre, and rating corresponding to our given arguments and return
+    else:
+        movie_details = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie_details
+
+# adds a movie object to a user's watched list
+# parameters: user_data - a user's movie information including watched list, movie - movie you have watched
+def add_to_watched(user_data, movie):
+    # we can append the movie object to the user's watched list
+    user_data["watched"].append(movie)
+    return user_data
+
+# adds a movie object to a user's want to watch list
+# parameters: user_data - a user's movie information including watched list, movie - movie you want to watch
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+# moves a movie object from a user's want to watch list to their watched list, different from add_to_watched() which adds a movie object not already in the want to watch list to the watch list
+def watch_movie(user_data, title):
+    # look for the movie in the want to watch list by looping through and matching the titles
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            # using the add_to_watched() function add the movie object to the watched list 
+            add_to_watched(user_data, movie)
+            # we can now remove the movie from the user's want to watch list
+            user_data["watchlist"].remove(movie)
+            # we return here and end the loop
+            return user_data
+    # the function will not add the move to the watched list if it was not found in the want to watch list
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# calculates a user's average rating for movies from their watched list (let's see how their taste is)
+# parameters: user_data - a user's movie information including watched list
+# returns float average rating, 0.0 if empty watched list
+def get_watched_avg_rating(user_data):
+    # set the avg rating to 0 initially, if they have no movies in their watched list this is the default value returned
+    average_rating = 0.0
+    # make sure that their watched list is not empty because we will later be dividing by the length of it (we dont want a 0 denominator)
+    if user_data["watched"]:

```

Comment:
There's actually not a need for this `if` statement. in the problem description we say that you can assume that if there are no movies the user have watched `user_data["watched"]` will be an empty list. Since a `for` loop over an empty list never enters its block, it would be fine, since `average_rating` would remain `0.0`.

However, if we did allow `user_data["watched"]` to be `None`, in that case, the `if` conditional would be acceptable to avoid an error.